### PR TITLE
Fix/Allow reading expired locked object

### PR DESCRIPTION
**Pull request by carpawell**
Related to https://github.com/TrueCloudLab/frostfs-api/issues/2.

### CODE REVIEW

**Review by acid-ant — DISMISSED** [review]

File: pkg/local_object_storage/metabase/exists.go [hunk 1]
```diff
@@ -100,6 +100,11 @@ func (db *DB) exists(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) (exists b
 //   - 2 if object is covered with tombstone;
 //   - 3 if object is expired.
 func objectStatus(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) uint8 {
+	// locked object could not be removed/marked with GC/expired
+	if objectLocked(tx, addr.Container(), addr.Object()) {
+		return 0
+	}
+
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/exists.go:107]: Can we have an explicit unit-test for this behaviour?

Comment (carpawell) [reply pkg/local_object_storage/metabase/exists.go:107]: sure, added

**Review by acid-ant — DISMISSED** [review]

File: pkg/local_object_storage/metabase/lock_test.go [hunk 2]
```diff
@@ -169,6 +169,27 @@ func TestDB_Lock(t *testing.T) {
...
+	// put an object
+	addr := putWithExpiration(t, db, object.TypeRegular, 124)
+
+	// expire the obj
+	es.e = 125
+	_, err := metaGet(db, addr, false)
+	require.ErrorIs(t, err, meta.ErrObjectIsExpired)
+
+	// lock the obj
+	require.NoError(t, db.Lock(addr.Container(), oidtest.ID(), []oid.ID{addr.Object()}))
```
Comment (fyrchik) [inline pkg/local_object_storage/metabase/lock_test.go:186]: Can we _lock_ an already expired object or is it only for tests here?

**Review by fyrchik — DISMISSED** [review]

File: pkg/local_object_storage/metabase/lock_test.go (hunk 2 above)
Comment (carpawell) [reply pkg/local_object_storage/metabase/lock_test.go:186]: > Can we lock an already expired object

yes, and i do not see any problems with that, you? in practice i think that is almost impossible if GC is configured to be relatively fast

Comment (fyrchik) [reply pkg/local_object_storage/metabase/lock_test.go:186]: Again, not for this PR.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**